### PR TITLE
feat: limit test duration should only apply to the OONI websites descriptor

### PR DESCRIPTION
**Pull request by aanorbel**
Closes https://github.com/ooni/probe-multiplatform/issues/279

### CODE REVIEW

File: composeApp/src/commonMain/kotlin/org/ooni/probe/domain/RunDescriptors.kt [hunk 1]
```diff
@@ -78,7 +78,10 @@ class RunDescriptors(
             .filterNot { it.allTests.isEmpty() }
 
     private suspend fun List<NetTest>.downloadUrlsIfNeeded(taskOrigin: TaskOrigin): List<NetTest> =
-        map { test -> test.copy(inputs = test.inputsOrDownloadUrls(taskOrigin)) }
+        map { test ->
+            val urls = test.inputsOrDownloadUrls(taskOrigin)
+            test.copy(inputs = urls, callCheckIn = urls?.isNotEmpty())
```
Comment (sdsantos) [inline composeApp/src/commonMain/kotlin/org/ooni/probe/domain/RunDescriptors.kt:83]: Should it be `calledCheckIn = inputs != urls` or something like that?

File: composeApp/src/commonMain/kotlin/org/ooni/engine/Engine.kt [hunk 2]
```diff
@@ -149,6 +150,20 @@ class Engine(
...
+        netTest: NetTest,
+        preferences: EnginePreferences,
+    ) = if (taskOrigin == TaskOrigin.AutoRun) {
+        MAX_RUNTIME_DISABLED
+    } else if (netTest.callCheckIn == true) {
+        preferences.maxRuntime?.inWholeSeconds?.toInt()?.let { maxRuntimePreference ->
+            if (maxRuntimePreference > 0) 30 + maxRuntimePreference else MAX_RUNTIME_DISABLED
+        } ?: MAX_RUNTIME_DISABLED
+    } else {
+        netTest.test.runtime(inputs = netTest.inputs).inWholeSeconds.toInt()
```
Comment (sdsantos) [inline composeApp/src/commonMain/kotlin/org/ooni/engine/Engine.kt:164]: Isn't this number for estimation? Ewe shouldn't be using it for max runtime should we? I though it was unlimited unless for WebConnectivity...

File: composeApp/src/commonMain/kotlin/org/ooni/engine/Engine.kt [hunk 3]
```diff
@@ -149,6 +150,20 @@ class Engine(
...
 
+    private fun maxRuntime(
+        taskOrigin: TaskOrigin,
+        netTest: NetTest,
+        preferences: EnginePreferences,
+    ) = if (taskOrigin == TaskOrigin.AutoRun) {
+        MAX_RUNTIME_DISABLED
+    } else if (netTest.callCheckIn == true) {
+        preferences.maxRuntime?.inWholeSeconds?.toInt()?.let { maxRuntimePreference ->
+            if (maxRuntimePreference > 0) 30 + maxRuntimePreference else MAX_RUNTIME_DISABLED
```
Comment (sdsantos) [inline composeApp/src/commonMain/kotlin/org/ooni/engine/Engine.kt:161]: Why are we adding 30 seconds to the preference time?

**Review by sdsantos — CHANGES_REQUESTED** [review]

File: composeApp/src/commonMain/kotlin/org/ooni/engine/Engine.kt (hunk 3 above)
Comment (aanorbel) [reply composeApp/src/commonMain/kotlin/org/ooni/engine/Engine.kt:161]: legacy migration

Comment (aanorbel) [reply composeApp/src/commonMain/kotlin/org/ooni/engine/Engine.kt:161]: https://github.com/ooni/probe-android/blob/master/app/src/main/java/org/openobservatory/ooniprobe/test/test/WebConnectivity.java#L50-L60

**Pull request merged**